assets/js/script.js

```diff
@@ -1,30 +1,28 @@
-let bookArray = [];
-const html = [];
-const myStorage = window.localStorage;
-bookArray = Object.entries(myStorage);
+class Book {
+  constructor() {
+    this.bookArray = Object.entries(window.localStorage);
+    this.html = [];
+  }
 
-// eslint-disable-next-line no-unused-vars
-function addBooks() {
-  const bookName = document.getElementById('name-book').value;
-  const bookAuthor = document.getElementById('author-book').value;
-  localStorage.setItem(bookName, bookAuthor);
-}
-
-function removeBooks(i) {
-  const title = i.firstChild.innerText;
-  Object.fromEntries(bookArray);
-  localStorage.removeItem(title);
-  bookArray = Object.entries(myStorage);
-  i.remove();
-}
+  addBooks() {
+    this.bookName = document.getElementById('name-book').value;
+    this.bookAuthor = document.getElementById('author-book').value;
+    localStorage.setItem(this.bookName, this.bookAuthor);
+  }
```

Comment:
Hi how are you, it happens that this is a feature o the LocalStorage itself, when I try to save an item with the same key, it will replace the value. Can you help me with this I don't know if I can do it different thanks

Reply:
Help Please

Reply:
Hello

Reply:
The easiest way to solve this is to prevent duplicate values. You can check if the book with the same name exists before creating a new one. The proper way is to give unique ids to each record and manipulating data with ids instead of the name.

For example:
```javascript
 addBook(title, author) {
    id = generateID(); // This can be index of book in array 
    this.books = this.books.concat({ title, author, id});
    return this.books; 
  }

  removeBook(bookID) {
     this.books = this.books.filter(({ id }) => bookID !== id); // remove book with the specific id
    return this.books; 
  }


```

Reply:
Thanks so much